### PR TITLE
added samrai package file

### LINKED ISSUE #5: write a .package file for SAMRAI
Our most important dependency. We also need to patch it with the patch attached to the latest IBAMR release.

### PR DESCRIPTION
**Pull request by brynbarker**
resolves #5

### CODE REVIEW

File: IBAMR-toolchain/packages/samrai.package [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+package_specific_setup () {
+    cd ${BUILDDIR}
+    cp -rf ${UNPACK_PATH}/${EXTRACTSTO}/* .
+
+    # make sure no other invalid PETSC_DIR is set:
+    unset SAMRAI_DIR
+    
+    ${PYTHON_INTERPRETER} ./configure --prefix=${INSTALL_PATH} ${CONFOPTS}
+    quit_if_fail "samrai ./configure failed"
```
Comment (drwells) [inline IBAMR-toolchain/packages/samrai.package:77]: SAMRAI uses autotools - not `BuildSystem`. This isn't going to work with SAMRAI.

We should probably hold off on merging this until we include the patch and can verify that this works on at least our machinse.

**Comment by drwells** [comment]
Does this actually work? I would prefer not to merge this until we have verified that it really does download and compile SAMRAI. We may need to write some more patches for autoibamr.sh first.

**Comment by brynbarker** [comment]
That's a good call. Let's hold off until we know it works with autoibamr.sh

**Comment by brynbarker** [comment]
@drwells this package file with the patch are working correctly now

**Pull request merged**